ow_power_system/src/PowerSystemNode.cpp

```diff
@@ -78,65 +80,97 @@ bool PowerSystemNode::loadSystemConfig()
+  if (filename.substr(filename.size() - 4) != ".csv")
+  {
+    throw ERR_INVALID_FILE_NAME;
+  }
+
   if (file.fail())
   {
-    cerr << "Unable to open data file" << filename << endl;
+    throw ERR_NO_CSV_FOUND;
+    return PrognoserVector();
   }
+
   // Skip header line
   file.ignore(numeric_limits<streamsize>::max(), '\n');
 
   auto now = system_clock::now();
 
   PrognoserVector result;
-  while (file.good())
+
+  // Line number starts at 2 instead of 1 because the first line is the header.
+  int line_number = 2;
+
+  try
+  {
+    while (file.good())
+    {
+      PrognoserMap data;
+      string line;
+      getline(file, line);
+      if (line.empty())
+        continue;
+
+      stringstream line_stream(line);
+      string cell;
+
+      // Confirm the line contains the expected number of columns.
+      auto cols = std::count(line.begin(), line.end(), ',') + 1;
+
+      ROS_INFO_STREAM("COLS: " << cols);
+
+      if (cols != CSV_EXPECTED_COLS)
+        throw ERR_CSV_FORMAT;
+
+      getline(line_stream, cell, ',');
+      double file_time = stod(cell);
+      auto timestamp = now + milliseconds(static_cast<unsigned>(file_time * 1000));
+
+      getline(line_stream, cell, ',');
+      Datum<double> power(stod(cell));
+      power.setTime(timestamp);
+
+      getline(line_stream, cell, ',');
+      Datum<double> temperature(stod(cell));
+      temperature.setTime(timestamp);
+
+      getline(line_stream, cell, ',');
+      Datum<double> voltage(stod(cell));
+      voltage.setTime(timestamp);
+
+      data.insert({ MessageId::Watts, power });
+      data.insert({ MessageId::Centigrade, temperature });
+      data.insert({ MessageId::Volts, voltage });
+
+      result.push_back(data);
+
+      line_number++;
+    }
+  }
+  catch(...) // Many possible different errors could result from reading an improperly formatted CSV.
   {
-    PrognoserMap data;
-    string line;
-    getline(file, line);
-    if (line.empty())
-      continue;
-
-    stringstream line_stream(line);
-    string cell;
-    getline(line_stream, cell, ',');
-    double file_time = stod(cell);
-    auto timestamp = now + milliseconds(static_cast<unsigned>(file_time * 1000));
-
-    getline(line_stream, cell, ',');
-    Datum<double> power(stod(cell));
-    power.setTime(timestamp);
-
-    getline(line_stream, cell, ',');
-    Datum<double> temperature(stod(cell));
-    temperature.setTime(timestamp);
-
-    getline(line_stream, cell, ',');
-    Datum<double> voltage(stod(cell));
-    voltage.setTime(timestamp);
-
-    data.insert({ MessageId::Watts, power });
-    data.insert({ MessageId::Centigrade, temperature });
-    data.insert({ MessageId::Volts, voltage });
-
-    result.push_back(data);
+    ROS_ERROR_STREAM("Failed to read " << csv << ": Improper formatting detected on line " << line_number << "." << endl << 
+                     "Confirm " << csv << " follows the exact format of example_fault.csv before retrying.");
```

Comment:
Please shorten these lines.  Our coding guidelines suggest 80 columns, with a max of 120 if shorter is not practical.